src/main/java/com/google/api/codegen/configgen/mergers/ConfigMerger.java

```diff
@@ -16,16 +16,23 @@
 
 import com.google.api.codegen.ConfigProto;
 import com.google.api.codegen.config.ApiModel;
+import com.google.api.codegen.configgen.ConfigHelper;
+import com.google.api.codegen.configgen.ConfigYamlReader;
+import com.google.api.codegen.configgen.MissingFieldTransformer;
+import com.google.api.codegen.configgen.NodeFinder;
 import com.google.api.codegen.configgen.nodes.ConfigNode;
 import com.google.api.codegen.configgen.nodes.FieldConfigNode;
+import com.google.api.codegen.configgen.nodes.ScalarConfigNode;
 import com.google.api.codegen.configgen.nodes.metadata.DefaultComment;
 import com.google.api.codegen.configgen.nodes.metadata.FixmeComment;
+import java.io.File;
 
 /** Merges the gapic config from an ApiModel into a ConfigNode. */
```

Comment:
Not part of this PR. But could you clarify the merging logic here? For example, is overriding allowed under current implementation?

Reply:
The merging for this pr checks if the default fields are present and adds them if not. The manual edits can add fields, change fields, but not remove default fields. I think we will need some sort of stateful merging to preserve deletions.